### PR TITLE
Bug in conversion of list/tuple

### PR DESCRIPTION
**Pull request by jtsoi**
@vbabiy 

There is a bug in serialization of list/tuple.

List is modified in place, which caused bugs later down the DRF flow.
Tuples cannot be modified at all, so the code throws exceptions if tuples are present.

Also added tests for these case.

Thanks @butschi for the first inflection based code.

### CODE REVIEW

File: djangorestframework_camel_case/util.py [hunk 1]
```diff
@@ -1,42 +1,80 @@
...
+    if isinstance(data, tuple):
+        return tuple(camelize(x) for x in data)
+
+    return data
 
 
 def underscoreize(data):
     if isinstance(data, dict):
-        new_dict = {}
+        new_dict = dict()
```
Comment (jakob-o) [inline djangorestframework_camel_case/util.py]: Shouldn't this be a OrderedDict as in camelize?

**Comment by jakob-o** [comment]
Hi,

thank you for the effort. This would make PR my https://github.com/vbabiy/djangorestframework-camel-case/pull/17 obsolete. I also like that there's some testcoverage! Is there any chance this will be merged soon @vbabiy? I'd also be glad to contribute some more testcases to improve the stability of this library, which - for me - is a must when using DRF.

Regards

Jakob

**Comment by jtsoi** [comment]
@jakob-o, thanks, =)
I don't think OrderedDict is needed.
We are basically overwriting the JSONRenderer/Parser:
https://github.com/tomchristie/django-rest-framework/blob/master/rest_framework/parsers.py#L67
Let me know if you disagree.

Not sure @vbabiy is actively maintaining this repo anymore. I was thinking maybe @tomchristie wants to fork this into the drf organisation. Must be a pretty common use case.

**Comment by jakob-o** [comment]
Hey,

in my opinion at least we should specify the behaviour. Since the renderer keeps the order I would expect the parser to do so too. Whilst you could argue it is more important on the renderer side, because developers tend to see the output more often then what they are parsing, it feels inconsistent. Client code however does not care about the key order. I'm also not sure about the performance impact of an OrderedDict vs a dict. Maybe we should consider a "keep_order" flag on the renderer... Not sure.

Regards

Jakob

**Comment by tomchristie** [comment]
> I was thinking maybe @tomchristie wants to fork this into the drf organisation. Must be a pretty common use case.

I've no problem with common use cases being catered for in third party repos where possible.
More than happy to consider any docs pull requests linking to this from appropriate points in our docs tho'.

**Comment by jakob-o** [comment]
@vbabiy Are you still maintaining this repo?

**Comment by dbrgn** [comment]
same question here :) I use this module in multiple projects, and it would be good if at least pull requests could be processed.

if @vbabiy doesn't have time anymore, maybe it could indeed be moved to the drf organization, @tomchristie?

**Comment by tomchristie** [comment]
We don't have any repos under a REST framework organization right now.
At the moment I'd suggest that someone would need to take this on independently. (eg. issue a djangorestframework-camel-case-2 to PyPI) we could then link to the new package/repo from the docs (assuming there's no movement/objection from here)

**Comment by jtsoi** [comment]
I am on parental leave, have some spare time. Will fork and submit a PR to
the Drf docs during next week.

/JT

**Comment by tomchristie** [comment]
> I am on parental leave

Congratulations! 😄 🎉

**Comment by vstoykov** [comment]
There is also another PR #27 which will also do the same and has less changes in it.

**Comment by vbabiy** [comment]
Sorry for the delay guys this should be fixed in master now. 

Thanks to @jtsoi for the PR

**Pull request closed without merging**